packages/material-ui-lab/src/Timeline/Timeline.tsx

```diff
@@ -21,12 +22,41 @@ export const styles = () => ({
   alignAlternate: {},
 });
 
-const Timeline = React.forwardRef(function Timeline(props, ref) {
+export type TimelineClassKey = keyof WithStyles<typeof styles>['classes'];
+
+export interface TimelineProps extends StandardProps<React.HTMLAttributes<HTMLUListElement>> {
+  /**
+   * The position where the timeline's content should appear.
+   * @default 'left'
+   */
+  align?: 'left' | 'right' | 'alternate';
+  /**
+   * The content of the component.
+   */
+  children?: React.ReactNode;
+  /**
+   * Override or extend the styles applied to the component.
+   */
+  classes?: {
+    /** Styles applied to the root element. */
+    root?: string;
+    /** Styles applied to the root element if `align="left"`. */
+    alignLeft?: string;
+    /** Styles applied to the root element if `align="right"`. */
+    alignRight?: string;
+    /** Styles applied to the root element if `align="alternate"`. */
+    alignAlternate?: string;
+  };
+  ref?: React.Ref<HTMLUListElement>;
```

Comment:
I couldn't use:

```suggestion
  ref?: React.Ref<unknow>;
```

which is the default

```suggestion
```

it fails with a "can't assign unknown to HTMLUListElement" error message if I don't set the correct type.

Reply:
The outer interface is more important here. I've gone through length to find the best possible type for `ref` so we shouldn't revert this partially just because the inner types are annoying. This is part of why I have problems with migrating to typescript: it might regress the DX for library users.

Reply:
https://github.com/microsoft/TypeScript/issues/30748 and https://github.com/mui-org/material-ui/pull/15199 for more context

Reply:
Ok, so we can simply `// @ts-expected-error`? I like that :)

Reply:
I wouldn't say you can do anything "simply" in TypeScripts' JSX domain :grinning: It's just bit unfortunate that this pattern of assigning and reading refs isn't supported by TS because we don't have control over covariance vs contravariance in function calls. But yes, you _probably_ can `@ts-expect-error` assignments of less specific refs to the `ref` prop.